### PR TITLE
fix(slider): expression changed after check

### PR DESCRIPTION
**Pull request by cal-smith**
Closes IBM/carbon-components-angular#582

### CODE REVIEW

File: src/slider/slider.component.ts [hunk 1]
```diff
@@ -146,6 +146,9 @@ export class Slider implements AfterViewInit, OnDestroy, ControlValueAccessor {
 
 		this._value = v;
 		this.slidAmount = this.convertToPx(v);
+		setTimeout(() => {
```
Comment (youda97) [inline src/slider/slider.component.ts]: Don't think you need the curly brackets. And maybe make it one line?

Comment (cal-smith) [reply src/slider/slider.component.ts]: Sure :+1:

**Comment by cal-smith** [comment]
Oh wow, yeah that's not good

**Comment by cal-smith** [comment]
Should be fixed @youda97 👍

File: src/slider/slider.component.ts [hunk 2]
```diff
@@ -74,7 +68,7 @@ import { ControlValueAccessor, NG_VALUE_ACCESSOR } from "@angular/forms";
 						#thumb
 						class="bx--slider__thumb"
 						tabindex="0"
-						[ngStyle]="{'left.%': getFractionComplete() * 100}"
+
```
Comment (youda97) [inline src/slider/slider.component.ts]: extra line

File: src/slider/slider.component.ts [hunk 3]
```diff
@@ -84,8 +78,9 @@ import { ControlValueAccessor, NG_VALUE_ACCESSOR } from "@angular/forms";
 						(click)="onClick($event)">
 					</div>
 					<div
+						#filledTrack
 						class="bx--slider__filled-track"
-						[ngStyle]="{transform: 'translate(0%, -50%)' + scaleX(getFractionComplete())}">
+						>
```
Comment (youda97) [inline src/slider/slider.component.ts]: should be on previous line

File: src/slider/slider.component.ts (hunk 2 above)
Comment (cal-smith) [reply src/slider/slider.component.ts]: Oh man I can't believe I forgot about these 🤦‍♂

**Comment by cal-smith** [comment]
Now that's super weird 🤔 I'll take a look ...

**Comment by cal-smith** [comment]
@youda97 fixed!

File: src/slider/slider.component.ts [hunk 4]
```diff
@@ -129,13 +122,30 @@ export class Slider implements AfterViewInit, OnDestroy, ControlValueAccessor {
 	/** Used to generate unique IDs */
 	private static count = 0;
 	/** The lower bound of our range */
-	@Input() min = 0;
+	@Input() set min(v) {
+		this._min = v;
+		this.value = this.value;
```
Comment (youda97) [inline src/slider/slider.component.ts]: Does this do anything?

Comment (cal-smith) [reply src/slider/slider.component.ts]: Yeah 😅 it forces the component to update ... I'll add a comment 👍

**Pull request merged**